test/bigfloat.jl

```diff
@@ -0,0 +1,9 @@
+@test big(2)^big(3) == 8
+
+for T in [Int8, UInt8, Int16, UInt16, Int32, UInt32, Int64, UInt64, Int128, UInt128, BigInt]
+	@test T(2)^big(3) == 8
```

Comment:
I already merged the original, so I'm going to keep the cherry-pick as close as possible and also pick the fixup, which also fixes the tests to actually test BigFloats.